Newsletters: add 174 (2021-11-10)

harding opened the pull request:
- [x] Bitcoin Core PR Review Club @glozow
- [x] `Rust-Lightning #1078` Implement channel_type negotiation @adamjonas
- [x] `Rust-Lightning #1144` Penalize failed channels @dongcarl
- [x] `BIPs #1215` Merge pull request #1215 from JeremyRubin/ctv-updates @jnewbery
- [x] Topics/releases @harding

jnewbery commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+Max Hillebrand (MaxHillebrand),
+Michael Folkson (michaelfolkson),
+Michael Ford (fanquake),
+nopara73,
```
```suggestion
Adam Ficsor (nopara73),
```

(from https://youtu.be/QiySI4-MWww?t=11, linked from https://gist.github.com/nopara73)

jnewbery commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+Alexandre_Chery,
+Alistair_Mann,
+amiti,
+_andrewtoth,
+andrewtoth,
```
Duplicates

jnewbery commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+averagepleb,
+bcman,
+belcher,
+benthecarman,
+benthecarman81,
```
Duplicates

jnewbery commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+evankaloudis,
+faketoshi,
+fanquake,
+fedorafan,
+fedorafansuper,
```
Probably duplicates

jnewbery commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+
+**Miner signaling**
+
+We also thank all of the miners since block FIXME that have signaled
```
Taproot's speedy trial nStartTime was April 24th:

```
        consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nStartTime = 1619222400; // April 24th, 2021
```

The first block for which the activation was STARTED state was therefore the next retarget block [681408](https://blockstream.info/block/000000000000000000020210114d7faf44223e1d620991e468af964514ff96e3?expand), mined on May 1st.

jnewbery reviewed (commented):
Looks good so far. I've added my write-up for BIPs 1215.

harding commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,157 @@
+      for their blocks, reducing the amount of data lightweight clients
+      need to download to monitor for discrepancies between the filters
+      being served by different peers.
+
```
How do you feel about that?  My personal feeling on the idea is "meh" and that's made me also meh on including it.  If you have any feeling for including it at all, I'm happy to write it up.

jnewbery replied:
I'm neutral. I haven't thought about it enough to form an opinion.

The fact that Laolu left it out of the original email does suggest that it's a bit of an afterthought, so I don't have any problem with leaving it out of the newsletter.

jnewbery commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+
+  q2="What does `CTxMemPool::check()` do? Whose responsibility is it to call
+    that function?"
+  a2="`CTxMemPool::check()` checks that each transaction's inputs correspond to
```
I don't think this makes it clear that that this is checking the entire contents of the mempool (and could instead be checking transactions that are being submitted to the mempool). How about something like:

```suggestion
  a2="`CTxMemPool::check()` does checks on _all_ the transactions currently in the mempool. It checks that each transaction's inputs correspond to
```

glozow replied:
thanks, changed to "checks that all transactions' inputs correspond to available UTXOs and performs internal consistency checks over the entire mempool."

jnewbery commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+  q1="Which components currently call `AcceptToMemoryPool`? Which of the ATMP calls are from external client code and which are from inside validation?"
+  a1="Excluding calls from tests, there are four call sites:
+  1. When the node starts, it [loads][atmp disk] transactions from mempool.dat and
+     calls ATMP to re-validate the transactions and restore mempool contents. This
+     call originates from a component 'external' to validation.
```
I disagree that this is external to validation. `LoadMempool()` is contained within validation.cpp and is called from a `CChainState` object (see https://bitcoincore.reviews/23173#l-107).

glozow replied:
I think this was a typo - thanks!

jnewbery commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+  minimum feerate is 3sat/vB, an individual transaction with a 1sat/vB feerate may
+  be accepted. ATMP is called with `bypass_limits` during a [reorg][atmp bypass
+  limits]; these transactions may have low individual feerates but high descendant
+  feerates. The size of the transactions is limited to
```
```suggestion
  feerates. The total size of all transactions to re-add to the mempool during a reorg is limited to
```

glozow replied:
fixed, thanks

jnewbery commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+- [Rust-Lightning #1078][] adds `channel_type` negotiation as defined in [BOLTs
+  #880][] and covered in [Newsletter #165][news165 bolts-880]. This
+  implementation does not currently send the proposed feature bit from [BOLTs
+  #906][], which are likely to be required for turbo channels, but if not, will
+  be needed for anchor channels.
```
BOLTs 906 isn't required for turbo channels/anchor outputs. When Matt says "It is likely to be required for turbo channels [...]", he's referring to support for BOLTs 880.

```suggestion
  #906][]. [BOLTs #880][] is required for [anchor channels][topic anchor outputs] and may also be needed to support [turbo channels](https://bitcoinops.org/en/newsletters/2021/07/07/#zero-conf-channel-opens).
```

jnewbery commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+
+- [Rust-Lightning #1144][] adds a penalization mechanism to the route scoring
+  logic. The penalty is applied to failed channels between payment retries to
+  inform the path finding algorithm of potentially faulty channels.
```
"pathfinding" is slightly more common in our repo than "path finding". I suggest that we standardize to pathfinding and add it to the style guide.

glozow commented on _posts/en/newsletters/2021-11-10-newsletter.md:
```diff
@@ -0,0 +1,225 @@
+  q0="What is `cs_main`? Why is it called `cs_main`?"
+  a0="`cs_main` is a mutex intended to synchronize multi-threaded access to
+validation state. In reality, it also guards non-validation data in
+net\_processing, and multiple contributors wish to minimize the usage of`cs_main`.
```
not sure, my typical style is to keep file names in plaintext, but i'm happy to go with whatever the convention is in bitcoinops?

glozow replied:
I changed this to "data used in P2P logic"

harding commented on _includes/specials/taproot/en/21-thanks.md:
```diff
@@ -0,0 +1,509 @@
+Andrew Poelstra,
+Anthony Towns,
+Antoine Riard,
+Ariel Lorenzo-Luaces,
+Ariel Luaces,
```
Confirmed, they always sign their name in the email body with the hyphenated form.

Xekyo reviewed (commented):
I'd like to propose that we generally separate value and unit of measurement by a (protected) space as per the SI standard: https://en.wikipedia.org/wiki/Space_(punctuation)#Unit_symbols_and_numbers

harding commented:
Pushed edits (and updates to the Style guide) for all feedback (I hope).  Thank you everyone for the reviews and suggestions!  Also reviewed @glozow @dongcarl @adamjonas @jnewbery contributions (thank you!) with a few tiny edits.

Updated releases/RCs and added topic links.

I'll try to check again tonight before bed for any final feedback, thank you everyone!

bitschmidty commented:
Looks great team. Thank you @harding @dongcarl @jnewbery @adamjonas and @glozow for authoring and @Xekyo for the 👓
